include/irr/asset/IMeshPacker.h

```diff
@@ -0,0 +1,123 @@
+    {
+        m_outVtxInputParams.enabledAttribFlags  = preDefinedLayout.enabledAttribFlags;
+        m_outVtxInputParams.enabledBindingFlags = preDefinedLayout.enabledBindingFlags;
+
+        memcpy(m_outVtxInputParams.attributes, preDefinedLayout.attributes, sizeof(m_outVtxInputParams.attributes));
+
+        //TODO: init allocators 
+        //m_alctrReservedSpace[0] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_vtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[0], 0u, 0u, 0u, allocParams.vertexBuffSupportedSizeInBytes / allocParams.vertexbufferMinAllocSize, allocParams.vertexbufferMinAllocSize);
+        
+        m_alctrReservedSpace[1] = static_cast<uint8_t*>(_IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<uint32_t>::reserved_size(alignof(uint16_t), allocParams.indexBuffSupportedSizeInBytes / sizeof(uint16_t), allocParams.indexBufferMinAllocSize), _IRR_SIMD_ALIGNMENT));
+        m_idxBuffAlctr = core::GeneralpurposeAddressAllocator<uint32_t>(m_alctrReservedSpace[1], 0u, 0u, alignof(uint16_t), allocParams.indexBuffSupportedSizeInBytes / sizeof(uint16_t), allocParams.indexBufferMinAllocSize);
+
+        //m_alctrReservedSpace[2] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_perInsVtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[2], 0u, 0u, 0u, allocParams.perInstanceVertexBuffSupportedSizeInBytes / allocParams.perInstanceVertexbufferMinAllocSize, allocParams.perInstanceVertexbufferMinAllocSize);
+
+        //init MDI data allocator
+        m_alctrReservedSpace[3] = static_cast<uint8_t*>(_IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<uint32_t>::reserved_size(alignof(uint32_t), allocParams.MDIDataBuffSupportedSizeInBytes / sizeof(MDIStructType), allocParams.MDIDataBuffMinAllocSize), _IRR_SIMD_ALIGNMENT));
+        m_MDIDataAlctr = core::GeneralpurposeAddressAllocator<uint32_t>(m_alctrReservedSpace[3], 0u, 0u, alignof(uint32_t), allocParams.MDIDataBuffSupportedSizeInBytes / sizeof(MDIStructType), allocParams.MDIDataBuffMinAllocSize);
+
+        //1 attrib enabled == 1 binding
+        for (uint16_t attrBit = 0x0001, location = 0; location < 16; attrBit <<= 1, location++)
+        {
+            if (m_outVtxInputParams.enabledAttribFlags & attrBit)
+            {
+                m_outVtxInputParams.bindings[location].stride = getTexelOrBlockBytesize(static_cast<E_FORMAT>(m_outVtxInputParams.attributes[location].format));
+                m_outVtxInputParams.bindings[location].inputRate = preDefinedLayout.bindings[preDefinedLayout.attributes[location].binding].inputRate;
+            }
+        }
+    }
+
+    virtual PackedMeshBufferData alloc(const MeshBufferType const* meshBuffer) = 0;
+    virtual void commit(/*TODO*/) = 0;
```

Comment:
you need to take the same arguments as `alloc` + the `PackedMeshBufferData` that `alloc` returned

Reply:
actually I'd take the `PackedMeshBufferData` as a reference (non `const` to allow for mutability, which would only be useful to free up some MDI space)

Reply:
arguments should be `(ReservedAllocationMeshBuffers& reserved, const Iterator begin, const Iterator end);`

upon success (the return of a valid `PackedMeshBufferData`) the `reserved` passed into the function shall be put into an invalid state (so that you do not try to `free` using it)